src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -96,12 +96,14 @@ internal interface IMessageSource
     internal abstract class HostEnvironmentBase<TEnv> : ChannelProviderBase, IHostEnvironment, IChannelProvider
         where TEnv : HostEnvironmentBase<TEnv>
     {
+        public bool IsCanceled { get; protected set; }
```

Comment:
Why is this moved from all the other property definitions?

Reply:
Didn't seem the need to include it there. The way I have it now is sufficient to cancel components.